src/libmeasurement_kit/ooni/collector_client.cpp

```diff
@@ -15,7 +15,7 @@ using namespace mk::report;
 
 static const std::regex re_name{"^[A-Za-z0-9._-]+$"};
 static const std::regex re_version{
-    "^[0-9]+.[0-9]+(.[0-9]+(-[A-Za-z0-9._-]+)?)?$"};
+    R"(^[0-9]+\.[0-9]+(\.[0-9]+(-[A-Za-z0-9._-]+)?(\+[0-9]+)?)?$)"};
```

Comment:
Can we make this regexp looser at this point? What is the purpose of having validation of it inside of MK?

In theory this is also a valid semver string:
```
0.1.1-alpha.2+23.b.a
```

Which that regexp would fail on.

Maybe stealing this regexp is a good idea: https://github.com/sindresorhus/semver-regex/blob/master/index.js

Reply:
> What is the purpose of having validation of it inside of MK?

Be strict in what you send, liberal in what you accept.

> Maybe stealing this regexp is a good idea: https://github.com/sindresorhus/semver-regex/blob/master/index.js

Let me see...

Reply:
> Be strict in what you send, liberal in what you accept.

Technically what you are doing here is the opposite, though, given the fact that this thing is given to you by the user of the library.

Reply:
So, I have imported the regex you suggested!